src/condor_contrib/bosco/bosco_cluster

```diff
@@ -289,17 +289,17 @@ ssh_detect_linux_distro () {
     os_release=`ssh $remote_host "cat /etc/os-release" 2> /dev/null`
     [[ $? -eq 0 ]] || return 1
 
-    dist=`echo "$os_release" | awk -F '=' '/^ID=/ {print $2}' | tr -d '"'`
+    dist=`echo "$os_release" | awk -F '=' '/^ID(_LIKE)*=/ {print $2}'`
```

Comment:
This is going to give you two lines on the rhel-likes; I don't know how that works with the case statement.

You should put the values of ID and ID_LIKE in separate variables (the second of which may be empty).

Reply:
Something like
```
    os_id=`echo "$os_release" | awk -F '=' '/^ID=/ {print $2}'`
    os_id_like=`echo "$os_release" | awk -F '=' '/^ID_LIKE=/ {print $2}'`
    ...
    if [[ $os_id_like = *rhel* ]]; then
        echo "CentOS${major_ver}"
    else
        case $os_id in
            debian)
                echo "Debian${major_ver}" ;;
            ubuntu)
                echo "Ubuntu${major_ver}" ;;
            *)
                return 1 ;;
        esac
    fi
```

Reply:
Case wildcards work seem to work fine over multiple lines

```
blin@blin-latitude:~$ podman run --rm -it rockylinux/rockylinux:8 /bin/bash
[root@a4de04e0a3b8 /]# dist=`awk -F '=' '/^ID(_LIKE)*=/ {print $2}' /etc/os-release`
[root@a4de04e0a3b8 /]#     case "$dist" in
>         (*rhel*|*centos*|*scientific*)
>             echo "CentOS${major_ver}" ;;
>         *ubuntu*)
>             echo "Ubuntu${major_ver}" ;;
>         *debian*)
>             echo "Debian${major_ver}" ;;
>     esac
CentOS
```

Reply:
OK. Well I like my more explicit version better -- I can foresee "wtf did this match" incidents occurring otherwise.

Reply:
One thing we'd lose with your version is auto-detection of Ubuntu/Debian variants but maybe that's fine.

Reply:
So, this distro detection is the remote host? right? I have not heard of any clusters that have deployed anything but stock Debian and Ubuntu on worker nodes. So, this may be OK, I would be nice to detect Debian and Ubuntu variants. But then you'd have to parse UBUNTU_CODENAME. I think that this is probably fine as is and we can augment it later if we discover variants on worker nodes.

Reply:
One thing I just thought of: could you add a comment to the case statement saying that order matters because Ubuntu has an ID_LIKE=debian?

Reply:
Done

Reply:
Tim, I noticed with Mint that their version number doesn't match the corresponding Ubuntu version number. For example, Mint 19.3 (tricia) is based on Ubuntu 18.04 (bionic). However, Mint has UBUNTU_CODENAME set to the matching Ubuntu release, so parsing UBUNTU_CODENAME is what you'd have to do anyway.

Reply:
Yeah, the debian dist ecosystem is a mess